### DISCUSSION TITLE
Unable to delete commit history of repository

### DISCUSSION THREAD

**Discussion by sachavieuxroy** [community/community#23903] [New to GitHub]
<p>I tried deleting the commit history of my repository on github. I found instuctions on doing this at the link below :</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://medium.com/codex/how-to-delete-your-repository-commit-history-on-github-4f6072936459">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/964545/181079912-67aae8e7-ddf6-4283-ac95-8795e2f65044.png" class="site-icon" width="32" height="32">

      <a href="https://medium.com/codex/how-to-delete-your-repository-commit-history-on-github-4f6072936459" target="_blank" rel="noopener nofollow ugc" title="07:39PM - 21 April 2021">Medium – 21 Apr 21</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:356/500;"><img src="https://user-images.githubusercontent.com/964545/181079913-75fe5835-285f-48f0-a45a-1272f65f8376.jpeg" class="thumbnail" width="356" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/1/21e6d29776200a50a993e6b9135c28a83bfed838_2_356x500.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/2/1/21e6d29776200a50a993e6b9135c28a83bfed838_2_534x750.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/1/21e6d29776200a50a993e6b9135c28a83bfed838.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/1/21e6d29776200a50a993e6b9135c28a83bfed838_2_10x10.png"></div>

<h3><a href="https://medium.com/codex/how-to-delete-your-repository-commit-history-on-github-4f6072936459" target="_blank" rel="noopener nofollow ugc">How to Delete Your Repository Commit History on GitHub</a></h3>

  <p>Typically, you should never delete your commit history. However, if you’re working on your own project or a repository that isn’t used by…</p>

  <p>
    <span class="label1">Reading time: 5 min read</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>To this date I have been unsuccessful. I was forced to make the repository private.</p>
<p>any help will be appreciated</p>

**Comment by bretonics** [comment]
<p>There are so many scenarios on deleting history; what exactly are you trying to do?</p>
<p>Deleting last 3 commits? Are they pushed to remote?</p>
<p>Deleting a specific commit back in history 5 commits back from HEAD?</p>
<p>Last I recall, <a href="https://sethrobertson.github.io/GitFixUm/fixup.html" rel="noopener nofollow ugc">this site</a> was useful walking someone through several scenarios. <em>Disclaimer</em> it’s been several years since I’ve tried this out, but it should be accurate.</p>
<p>Hope that helps!</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="sachavieuxroy" data-post="1" data-topic="196936">
<div class="title">
<div class="quote-controls"></div>
 sachavieuxroy:</div>
<blockquote>
<p>I tried deleting the commit history of my repository on github.</p>
</blockquote>
</aside>
<p>As <a class="mention" href="/bretonics">@bretonics</a> mentioned, there are many ways to go about.</p>
<p>I personally find that the simplest way is apply all the changes you want and then squash all commits together, which results in the repository history being re-written as there was ever only just on commit, you then only need to <code>git push --force</code> and the new history will be forced on the GitHub repository — beware that all previous commits will be lost with this operation, so consider it an irreversible operation (although there are ways to undo this via the Reflog).</p>
<p>As the article you linked mentions, rewriting history on the main branch of a repository is usually a bad idea.</p>
<p>If by “deleting” you mean resetting the repository to a pristine state (as if it was a new one, without history), then you only need to delete all contents (including the <code>.git/</code> folder), execute <code>git init</code>, add again the remote pointing to your GitHub origin, and force push — this way it would be as you had just created the repository now.</p>

**Comment by sachavieuxroy** [comment]
<p>Deleting the complete commit history of the repository except for the very last commit.</p>
<p>There currently 16 commits total in the repository.</p>

**Comment by kingthorin** [comment]
<p>Just do an interactive rebase of the whole history and make them all fix ups to a single commit. Then force push it.</p>

**Comment by sachavieuxroy** [comment]
<p>can you give me an example on how to do this?</p>

**Comment by kingthorin** [comment, marked as answer]
<p>Here’s some docs that should help, the first one is eclipse/eGit specific but may still help you to understand the process:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://www.instructables.com/How-to-Rebase-Interactive-With-Eclipse-eGit/">
  <header class="source">
      <img src="https://www.instructables.com/assets/img/siteassets/favicon.ico" class="site-icon" width="" height="">

      <a href="https://www.instructables.com/How-to-Rebase-Interactive-With-Eclipse-eGit/" target="_blank" rel="noopener nofollow ugc">Instructables</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/7570458/181079920-fe90a371-e9e0-44f8-8b0b-b4a616085ba0.png" class="thumbnail onebox-full-image" width="60" height="50">

<h3><a href="https://www.instructables.com/How-to-Rebase-Interactive-With-Eclipse-eGit/" target="_blank" rel="noopener nofollow ugc">How to Rebase Interactive With Eclipse (eGit)</a></h3>

  <p>How to Rebase Interactive With Eclipse (eGit): 20180718 - I "think" I've sorted out the issue with the images. If they appear zoomed in or don't make sense, try clicking. If you encounter an issue fire me a message and I'll try to help out.This...</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<aside class="onebox allowlistedgeneric" data-onebox-src="https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d">
  <header class="source">

      <a href="https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d" target="_blank" rel="noopener nofollow ugc">hackernoon.com</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/7570458/181079921-b8950930-57ac-4e62-8eff-3c16eff32e58.png" class="thumbnail" width="690" height="345" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/f/5fd6ecb4ef6b6e8b712df587766ead01ad97d2ae_2_10x10.png"></div>

<h3><a href="https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d" target="_blank" rel="noopener nofollow ugc">Beginner’s Guide to Interactive Rebasing | Hacker Noon</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p><a href="https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History" class="onebox" target="_blank" rel="noopener nofollow ugc">https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History</a></p>
